src/app/reporting/Engine.cpp

```diff
@@ -266,18 +266,26 @@ CHIP_ERROR CheckEventValidity(const ConcreteEventPath & path, const SubjectDescr
     err = provider->EventInfo(path, eventInfo);
     if (err != CHIP_NO_ERROR)
     {
-        outStatus = StatusIB(err);
-        return CHIP_NO_ERROR;
+        // cannot get event data to validate. Event is not supported.
+        // we still fall through into "ValidateClusterPath" to try to return a `better code`
+        // (i.e. say invalid endpoint or cluser), however if path seems ok we will
+        // return unsupported event as we failed to get event metadata.
+        outStatus = StatusIB(Status::UnsupportedEvent);
     }
+    else
+    {
+        // set up the status as "OK" as long as validation below works
+        outStatus = StatusIB(Status::Success);
 
-    requestPath.entityId = path.mEventId;
+        requestPath.entityId = path.mEventId;
 
-    err = GetAccessControl().Check(subjectDescriptor, requestPath, eventInfo.readPrivilege);
-    if (IsTranslatableAclError(path, err, outStatus))
-    {
-        return CHIP_NO_ERROR;
+        err = GetAccessControl().Check(subjectDescriptor, requestPath, eventInfo.readPrivilege);
```

Comment:
So this is still changing behavior from what we used to have (but not changing it to what the spec says, because what the spec says is not implementable), right?  Our current behavior is that ValidateClusterPath is first, then everything else.

Why are we doing that?  Why not keep the existing behavior until the spec gets its act together, so we only have _one_ behavior change for clients to deal with?

Reply:
What current code does is call validation and construct a status IB ... but then do ACL check and potentially construct another statusib:

![image](https://github.com/user-attachments/assets/c550e79b-764e-481c-8851-c816a8b52e25)


Then TestACLEvent does remember the last event

![image](https://github.com/user-attachments/assets/4c9bd7e2-15bc-468d-a545-d8abde855cec)

And our test for ACL checks for EXPECT_TRUE (and 2 is just as true as 1 ... fun...)

![image](https://github.com/user-attachments/assets/8126ceb1-f563-41b1-a60f-b00f49ce2d05)

So we actually get 2 StatusIB but never detect it:

![image](https://github.com/user-attachments/assets/57aaec80-5885-49c0-baa4-71aeb0fe7a2d)

After my changes, this is actually 1....

Reply:
> but then do ACL check and potentially construct another statusib:

That's clearly a bug, caused by a missing return....

Reply:
OK, so I guess the issue is the tests are testing something spec-like (but not the unimplementable spec) and you did not want to change the tests when you fixed the core "two statuses" bug...  I can live with the behavior change, then, but someone needs to own getting the spec fixed and then getting this code to implement the spec.  Because again: what this code is now doing is NOT what the spec currently says to do.

Reply:
I would expect https://github.com/project-chip/connectedhomeip/issues/37342 to be the actual "lets do what the spec says we should do" fix. Updates here are driveby to remove some of the more obvious issues (double statusib is crazy) however my intent was really "just" decoupling.